app/models/piece.rb

```diff
@@ -1,4 +1,11 @@
 class Piece < ApplicationRecord
-  belongs_to :player
+  belongs_to :player, class_name: 'Player', optional: true
   belongs_to :game
+
+  # we should not name a column "type", but this gets around the problem
+  # http://apidock.com/rails/ActiveRecord/Base/inheritance_column/class
+  # disable STI
+  self.inheritance_column = :_type_disabled
```

Comment:
Can remove these lines. Now that we have the STI branch merged in. If you set the type to one of the correct class names. Should work.